src/browser/BrowserAction.cpp

```diff
@@ -498,6 +501,29 @@ QJsonObject BrowserAction::handleGetTotp(const QJsonObject& json, const QString&
     return buildResponse(action, message, newNonce);
 }
 
+QJsonObject BrowserAction::handleGlobalAutoType(const QJsonObject& json, const QString& action)
+{
+    const QString nonce = json.value("nonce").toString();
+    const QString encrypted = json.value("message").toString();
+    const QJsonObject decrypted = decryptMessage(encrypted, nonce);
+
+    if (decrypted.isEmpty()) {
+        return getErrorReply(action, ERROR_KEEPASS_CANNOT_DECRYPT_MESSAGE);
+    }
+
+    QString command = decrypted.value("action").toString();
+    if (command.isEmpty() || command.compare("request-autotype") != 0) {
+        return getErrorReply(action, ERROR_KEEPASS_INCORRECT_ACTION);
+    }
+
+    const auto topLevelDomain = decrypted.value("search").toString();
```

Comment:
Let's limit the search text size to 256 characters, part of our sanitization push.